### PR TITLE
Add filters

### PR DESCRIPTION
**Pull request by torztomasz**
Resolves L2B-2596, L2B-2475, L2B-2720

### <samp>🤖 Generated by Copilot at 68764fe</samp>

This pull request adds functionality of project filters for tables and charts. It adds new components for empty and error states, a header with interaction zone, and a slide card for the rich select component. It also simplifies and enhances the proxy setup, the chart data format and urls, the radio group and checkbox components, and the chart range and currency controls. It updates the descriptions and display names of some layer2s and adds a feature to show the number of items for each tab.

### <samp>🤖 Generated by Copilot at 68764fe</samp>

*  Update the descriptions of degate and loopring layer2s and capitalize the display name of loopring ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-e13c696fd3ef337833faaf32324b548ae7770ca1e301bbcbdd36f38b334f8cb8L60-R60), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-091d0279e1d0e492c30b0c418a24d5b9f6f22b581b1d364c8456548e5f0568c8L57-R57), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-cc9ae090378b70a1fbbd3a5e036f8b5a3a6452611a2eb8554f55d8b6cb453e47L46-R46), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-1f7a2e5da75342fa926dd3038a8dd8f596e96c0c30df901b600ae0830036f2caL82-R82))
*  Simplify the chart data format and urls and remove unused lodash at function from `createApi.ts` ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL11), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL26-R35), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-025853d2b2edb6af6003d1eaa0299bd54ad5ecbeea5fe01b2e30fc06b7272e7cL72-R61))
*  Add new components for chart header, empty state, and error state and render them in `Chart.tsx` based on the chart type and state ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1L6-R11), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1R37), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1R61), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1R68), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1R97-R103), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1L98-R117), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-51b78eb131f51c468e0c5ed4af1253cdbfcaaf21963df291603a9ed7bc21b3cfR1-R15), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-dbc0f07d46312d5f419d2e1cd7ba3371f787b979b00d340fe3c35c337c4c7f09R1-R21))
*  Add new props for custom classes and data attributes to `Checkbox.tsx`, `RadioGroup.tsx`, `RichSelect.tsx`, `PercentChange.tsx`, and `Tabs.tsx` and use them to customize the appearance and behavior of the chart controls and filters ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-50342caa690b7c1267f31f2c9ac4d96741f9e843b54b6a6d8b5d71d571c98f7bL1-R1), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-50342caa690b7c1267f31f2c9ac4d96741f9e843b54b6a6d8b5d71d571c98f7bL8-R12), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-50342caa690b7c1267f31f2c9ac4d96741f9e843b54b6a6d8b5d71d571c98f7bL16-R23), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-50342caa690b7c1267f31f2c9ac4d96741f9e843b54b6a6d8b5d71d571c98f7bL23-R29), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-50342caa690b7c1267f31f2c9ac4d96741f9e843b54b6a6d8b5d71d571c98f7bR37-R38), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-50342caa690b7c1267f31f2c9ac4d96741f9e843b54b6a6d8b5d71d571c98f7bL37-R45), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-e4efdfb4005aedf554fbbf4533d1638237ce1d8a6644b03e8ed0f8435b0e9e9fL8-R16), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-e4efdfb4005aedf554fbbf4533d1638237ce1d8a6644b03e8ed0f8435b0e9e9fR28), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL12-R12), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL25-R25), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL35), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL44-R42), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL51-R66), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL72-R77), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b2e616512354cbdf2ad01f536513419babb1dce379c4b072a036d5acf756a9cdL86-R90), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-8fa6a7c191759ac81298a5a6e827613d4d1a3595d48f6668561729ead96112ceR14), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-8fa6a7c191759ac81298a5a6e827613d4d1a3595d48f6668561729ead96112ceL24-R39), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-36ac1a11e4890685e93e117e0d9d7da2ede84afbd612f5d767e819dd9ec467f2R10-R11), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b5eec5cf3d1df516a14fc38038e7dbc0602dceec50db0f962784be6df06f0e2aR7-R8), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b5eec5cf3d1df516a14fc38038e7dbc0602dceec50db0f962784be6df06f0e2aR23), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b5eec5cf3d1df516a14fc38038e7dbc0602dceec50db0f962784be6df06f0e2aL26-R33), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-57ae00591d3e88873a121c03c307cdac9a8b5e18009bbbc6ee69853e7a95dcaaR10-R11), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-57ae00591d3e88873a121c03c307cdac9a8b5e18009bbbc6ee69853e7a95dcaaL15-R34))
*  Refactor the proxy middleware configuration in `gulpfile.js` to use a loop over an array of proxyUrls instead of hardcoding each url ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-684bea57696a81884bb6ad5fb0e9e04519cf6102b2592b18bdaf6762b5f92998R71-R76), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-684bea57696a81884bb6ad5fb0e9e04519cf6102b2592b18bdaf6762b5f92998L89-R104))
*  Modify the chart label, loader, and canvas elements to use the group-data attribute to control their opacity and interactivity based on the chart state ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-68afe544a4c5715274c5c44241a08da2f2e23995768c7f8ea1e21326833a6e4eL30-R30), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-73b9abfff8cf4526b09c37c3a96225ad09118fe970b10200ff8cd85ac4de19e9L7-R7), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-aa97966834e46135486f7af43f458eaf24fe88005046e7845f46ff767ed3abd1R97-R103))
*  Remove the className prop and the max-width classes from `EthereumActivityToggle.tsx` as they are no longer needed after the chart header redesign ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b263e0632d29c00af35007992f33c20c7e22b39b645724dffe6d13025ec11c0aL8), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-b263e0632d29c00af35007992f33c20c7e22b39b645724dffe6d13025ec11c0aL15))
*  Replace the WarningIcon component with the RoundedWarningIcon component in `TVLBreakdown.tsx` to match the design mockups ([link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-7ac5e3ed0444af27b3276c8a0e4cd417bc773eef391b303ddcfa15e6d5fa417dL3-R3), [link](https://github.com/l2beat/l2beat/pull/1932/files?diff=unified&w=0#diff-7ac5e3ed0444af27b3276c8a0e4cd417bc773eef391b303ddcfa15e6d5fa417dL45-R48))

Co-authored-by: Antoni Pawlak <antoni.pawlak@pm.me>

### CODE REVIEW

File: packages/frontend/src/components/table/filters/ScalingFilters.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const providers = uniq(items.map((i) => i.provider))
+    .sort()
+    .map((p) => ({
+      label: p ?? 'No provider',
+      value: generateSlugList(items, (i) => i.provider === p),
+    }))
+
+  const stages = uniq(items.map((i) => i.stage?.stage))
+    .sort()
+    .filter(noUndefined)
```
Comment (maciekzygmunt) [inline packages/frontend/src/components/table/filters/ScalingFilters.tsx:24]: Use `notUndefined` exported from shared-pure - packages/shared-pure/src/utils/notUndefined.ts

File: packages/frontend/src/components/table/filters/ScalingFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    case 'UnderReview':
+      return 'Under review'
+    default:
+      return stage
+  }
+}
+
+function noUndefined<T>(x: T | undefined): x is T {
+  return x !== undefined
+}
```
Comment (maciekzygmunt) [inline packages/frontend/src/components/table/filters/ScalingFilters.tsx:72]:
```suggestion
```

**Comment by antooni** [comment]
Provide description: context, most important changes, how to test

**Comment by antooni** [comment]
select stage on activity does not work

**Comment by antooni** [comment]
activity on branch preview does not work 
image

**Pull request merged**